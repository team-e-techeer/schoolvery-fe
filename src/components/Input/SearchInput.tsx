import colors from '@/constants/colors';
import { css } from '@emotion/react';
import { Link } from 'react-router-dom';
import { BiSearch as SearchIcon } from 'react-icons/bi';
import { FiArrowLeft as LeftIcon } from 'react-icons/fi';
interface Props {
  isLinking: boolean;
  onSubmit?: (e: React.FormEvent<HTMLFormElement>) => void;
  inputValue?: string;
  setInputValue?: React.Dispatch<React.SetStateAction<string>>;
  setFocus?: React.Dispatch<React.SetStateAction<boolean>>;
  searchTestId?: string;
  formTestId?: string;
}

const inputStyle = css`
  display: flex;
  width: 80%;
  margin: 0 auto;
  align-items: center;
  margin-bottom: 1rem;
  background-color: ${colors.grey100};
  border-radius: 5px;
  height: 5rem;
  padding: 1rem;
  align-content: center;
`;

const inputWithoutPrev = css`
  display: flex;
  width: 90%;
  margin: 3rem auto;
  margin-bottom: 1rem;
  background-color: ${colors.grey100};
  border-radius: 5px;
  height: 5rem;
  padding: 1rem;
  align-content: center;
  align-items: center;
`;

export default function Input({
  isLinking,
  onSubmit,
  inputValue,
  setInputValue,
  setFocus,
  searchTestId,
  formTestId,
}: Props) {
  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => setInputValue && setInputValue(e.target.value);
  return isLinking ? (
    <Link css={inputWithoutPrev} to="/search">
      <SearchIcon size={25} color={colors.mainColor} />
      <input
        css={css`
          margin-left: 1rem;
          background-color: inherit;
        `}
        placeholder="검색어를 입력해 주세요"
        type="text"
      />
    </Link>
  ) : (
    <div
      css={css`
        display: flex;
        align-items: center;
        margin-top: 2rem;
      `}
    >
      <Link
        to="/"
        css={css`
          cursor: pointer;
          margin-left: 5%;
        `}
      >
        <LeftIcon color={colors.grey800} size={30} />
      </Link>
      <form css={inputStyle} onSubmit={onSubmit} data-testid={formTestId}>
        <SearchIcon size={25} color={colors.mainColor} />
        <input
          css={css`
            margin-left: 1rem;
            background-color: inherit;
            width: 100%;
          `}
          placeholder="검색어를 입력해 주세요"
          type="text"
          id="search-input"
          autoFocus={true}
          value={inputValue}
          onChange={onChange}
          data-testid={searchTestId}
          onFocus={() => {
            setFocus && setFocus(true);
          }}
        />
      </form>
    </div>
  );
}
