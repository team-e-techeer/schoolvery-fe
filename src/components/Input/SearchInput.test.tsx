import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import SearchInput from './SearchInput';
describe('로그인 페이지 테스트', () => {
  it('렌더링 테스트', () => {
    const { container } = render(<SearchInput isLinking={true} />, { wrapper: BrowserRouter });
    expect(screen.getByPlaceholderText('검색어를 입력해 주세요')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('입력 테스트', () => {
    const { container } = render(<SearchInput isLinking={false} />, { wrapper: BrowserRouter });

    const input = screen.getByPlaceholderText('검색어를 입력해 주세요') as HTMLInputElement;
    expect(input).toBeInTheDocument();
    fireEvent.change(input, { target: { value: '123' } });
    expect(input.value).toBe('123');
    expect(container).toMatchSnapshot();
  });
});
