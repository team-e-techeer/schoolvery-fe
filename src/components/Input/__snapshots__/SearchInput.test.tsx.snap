// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`로그인 페이지 테스트 렌더링 테스트 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 90%;
  margin: 3rem auto;
  margin-bottom: 1rem;
  background-color: #f2f4f6;
  border-radius: 5px;
  height: 5rem;
  padding: 1rem;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  margin-left: 1rem;
  background-color: inherit;
}

<div>
  <a
    class="emotion-0"
    href="/search"
  >
    <svg
      color="#fc775a"
      fill="currentColor"
      height="25"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(252, 119, 90);"
      viewBox="0 0 24 24"
      width="25"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 18a7.952 7.952 0 0 0 4.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0 0 18 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
      />
    </svg>
    <input
      class="emotion-1"
      placeholder="검색어를 입력해 주세요"
      type="text"
    />
  </a>
</div>
`;

exports[`로그인 페이지 테스트 입력 테스트 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 2rem;
}

.emotion-1 {
  cursor: pointer;
  margin-left: 5%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 80%;
  margin: 0 auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 1rem;
  background-color: #f2f4f6;
  border-radius: 5px;
  height: 5rem;
  padding: 1rem;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
}

.emotion-3 {
  margin-left: 1rem;
  background-color: inherit;
  width: 100%;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <svg
        color="#333d4b"
        fill="none"
        height="30"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        style="color: rgb(51, 61, 75);"
        viewBox="0 0 24 24"
        width="30"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="19"
          x2="5"
          y1="12"
          y2="12"
        />
        <polyline
          points="12 19 5 12 12 5"
        />
      </svg>
    </div>
    <form
      class="emotion-2"
    >
      <svg
        color="#fc775a"
        fill="currentColor"
        height="25"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(252, 119, 90);"
        viewBox="0 0 24 24"
        width="25"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 18a7.952 7.952 0 0 0 4.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0 0 18 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
        />
      </svg>
      <input
        class="emotion-3"
        id="search-input"
        placeholder="검색어를 입력해 주세요"
        type="text"
        value=""
      />
    </form>
  </div>
</div>
`;
